src/main/scala/com/fulcrumgenomics/util/miseq/Sample.scala

```diff
@@ -71,5 +73,26 @@ class Sample(val sampleOrdinal: Int,
       case Some(str) => Some(str.trim)
     }
   }
+
+  /** Sets the sample barcode based on the read structure.
+    *
+    * @param sampleBarcodeReadStructures a read structure per read (ex. i7 and i5) that contains only sample barcode segments.
+    */
+  def setSampleBarcode(sampleBarcodeReadStructures: Seq[ReadStructure]): this.type = {
```

Comment:
This name is really confusing.  It's _updating_ the existing set of barcode bases, based on what we think we actually sequenced given the read-structure.  Maybe call it `updateSampleBarcodeBasedOnReadStructure`?